src/overlays/actors/ovl_En_Fr/z_en_fr.c

```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+    f32 xDistFromLink;
+    f32 zDistFromLink;
+    f32 yDistFromLink;
+
+    xDistFromLink = player->actor.posRot.pos.x - this->actor.posRot.pos.x;
+    zDistFromLink = player->actor.posRot.pos.z - this->actor.posRot.pos.z;
+    yDistFromLink = player->actor.posRot.pos.y - this->actor.posRot.pos.y;
+
+    return ((SQ(xDistFromLink) + SQ(zDistFromLink)) <= SQ(30.0f)) && (yDistFromLink >= 0.0f);
+}
+
+void EnFr_DecrementBlinkTimer(EnFr* this) {
+    if (this->blinkTimer) {
+        this->blinkTimer--;
+    } else {
+        this->blinkFunc = EnFr_DecrementBlinkTimerUpdate;
+    }
+}
+
+void EnFr_DecrementBlinkTimerUpdate(EnFr* this) {
+    if (this->blinkTimer) {
+        this->blinkTimer--;
+    } else if (this->eyeTexIndex) {
+        this->eyeTexIndex = 0;
+        this->blinkTimer = (s16)(Math_Rand_ZeroFloat(60.0f) + 20.0f);
+        this->blinkFunc = EnFr_DecrementBlinkTimer;
+    } else {
+        this->eyeTexIndex = 1;
+        this->blinkTimer = 1;
+    }
+}
+
+void EnFr_InitJumpingOutOfWater(EnFr* this, GlobalContext* globalCtx) {
+    // Frogs jumping out one at a time in order sTimerJumpingOutOfWater
+    if (sEnFrPointers.flags == sTimerJumpingOutOfWater[this->actor.params - 1]) {
+
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_060007BC, 1.0f, 0.0f,
+                             SkelAnime_GetFrameCount(&D_060007BC), 2, 0.0f);
+
+        EnFr_DrawActive(this);
+
+        this->actionFunc = EnFr_JumpingOutOfWater;
+    }
+}
+
+void EnFr_JumpingOutOfWater(EnFr* this, GlobalContext* globalCtx) {
+    Vec3f vec1;
+    Vec3f vec2;
+
+    // Increment flag to time the next frog jumping out of water
+    if (this->skelAnime.animCurrentFrame == 6.0f) {
+        sEnFrPointers.flags++;
+        this->skelAnime.animPlaybackSpeed = 0.0f;
+
+        // Midway through jump
+    } else if (this->skelAnime.animCurrentFrame == 3.0f) {
+        this->actor.gravity = -10.0f;
+        this->actor.speedXZ = 0.0f;
+        this->actor.velocity.y = 47.0f;
+    }
+
+    vec1.y = 0.0f;
+    vec1.x = 0.0f;
+    vec1.z = this->xzDistToLogSpot;
+
+    Matrix_RotateY(((this->actor.posRot.rot.y + 32768) / 32768.0f) * M_PI, MTXMODE_NEW);
+    Matrix_MultVec3f(&vec1, &vec2);
+
+    this->actor.posRot.pos.x = this->posLogSpot.x + vec2.x;
+    this->actor.posRot.pos.z = this->posLogSpot.z + vec2.z;
+
+    if (this->skelAnime.animCurrentFrame >= 3.0f) {
+        Math_SmoothScaleMaxF(&this->xzDistToLogSpot, 0.0f, 1.0f, 10.0f);
+    }
+
+    // The frog has reached the log
+    if (EnFr_IsBelowLogSpot(this, &vec2.y)) {
+        this->actor.gravity = 0.0f;
+        this->actionFunc = EnFr_OrientOnLogSpot;
+        this->unusedFloat = 0.0f;
+    }
+
+    if ((this->actor.velocity.y <= 0.0f) && (vec2.y < 40.0f)) {
+        this->skelAnime.animPlaybackSpeed = 1.0f;
+    }
+}
+
+void EnFr_OrientOnLogSpot(EnFr* this, GlobalContext* globalCtx) {
+    s16 func;
+    func = Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->actor.yawTowardsLink, 2, 10000, 100);
```

Comment:
after you merge master youll see a comment on this function that describes the purpose of the return value. itd be better to name it after that, cause `func` can be pretty confusing.
also, this is a case where id assign with the declare to put it on the same line